vcd/resource_vcd_nsxt_ipsec_vpn_tunnel.go

```diff
@@ -288,17 +288,11 @@ func resourceVcdNsxtIpSecVpnTunnelCreate(ctx context.Context, d *schema.Resource
 	d.SetId(createdIpSecVpnConfig.NsxtIpSecVpn.ID)
 
 	// Check if Tunnel Profile has custom settings and apply them
-	if _, isSet := d.GetOk("security_profile_customization"); isSet {
-		tunnelProfileConfig, err := getNsxtIpSecVpnProfileTunnelConfigurationType(d)
-		if err != nil {
-			return diag.Errorf("[nsx-t ipsec vpn tunnel create] error getting NSX-T IPsec VPN Tunnel Profile: %s", err)
-		}
-
-		_, err = createdIpSecVpnConfig.UpdateTunnelConnectionProperties(tunnelProfileConfig)
-		if err != nil {
-			return diag.Errorf("[nsx-t ipsec vpn tunnel create] error setting VPN Tunnel Profile: %s", err)
-		}
+	tunnelProfileConfig := getNsxtIpSecVpnProfileTunnelConfigurationType(d)
 
+	_, err = createdIpSecVpnConfig.UpdateTunnelConnectionProperties(tunnelProfileConfig)
+	if err != nil {
+		return diag.Errorf("[nsx-t ipsec vpn tunnel create] error setting VPN Tunnel Profile: %s", err)
```

Comment:
Should we not trigger the `TunnelConnectionProperties` if we can reliably identify that it is not set? (using `_, isSet := d.GetOk("security_profile_customization")` like in https://github.com/vmware/terraform-provider-vcd/pull/1072/files#diff-aadc80442709e37c8a86461fb33847073473b847e213ecdfb209f1c0a036eb06R331

That way we could avoid unnecessary API call at all

Reply:
Good call. Updated.